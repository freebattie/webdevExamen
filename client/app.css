*{
    margin: 0;
    padding: 0;
    border: 0;
    box-sizing: border-box;

    font-family: 'Fira Sans', sans-serif;
}

.menu{
    margin-top: 40px;
    padding: 100px;
    display: flex;

    border: 1px solid black;
    flex-wrap: wrap;
    justify-content: center;

}
.dish{
    margin: 10px;
    width: 300px;
    border: solid 2px black;
    height: 200px;
}
footer{
    width: 80px;
}
.editItem{
    display: flex;
    width: 300px;
    flex-wrap: wrap;
}
.button{
    color: black;
    padding: 5px 2px;
    text-align: center;
    text-decoration: none;
    display: inline-block;
    cursor:pointer;
    margin: 10px;
    font-size: 16px;
    background-color: white;

    border: 2px solid black;
    width: 70px;
    border-radius: 5px;
    transition-duration: 0.4s;
}
.menuheader{
    margin-bottom: 20px;
}
.orders{
    display: flex;
    flex-wrap: wrap;
}
.order{

    height: 300px;
    border: 2px black solid;
    width: 300px;
    margin: 10px;
}
.items{
    width: 100px;
    margin: 10px;
}
.ordersheader{
    width: 100%;
}
.editform{
    width: 400px;
    border: 2px solid black;
}
.itemsToEdit{
    width: 100%;
    display: flex;
    flex-wrap: wrap;

}