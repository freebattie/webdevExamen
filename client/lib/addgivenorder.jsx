export async function addGivenOrder(newOrder, setError, navigate, addOrder) {
  if (newOrder.orders.length < 1) {
    setError("No items in list");
    navigate("/error");
  } else {
    try {
      const test = await addOrder(newOrder);

      navigate("/Myorders");
    } catch (e) {
      setError(e);
      navigate("/error");
    }
  }
}
