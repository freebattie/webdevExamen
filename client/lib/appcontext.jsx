import React from "react";
import { fetchJSON, postJSON } from "./http.js";

export const Appcontext = React.createContext({
  async listDishes() {
    return await fetchJSON("/api/dish");
  },

  async fetchLogin() {
    return await fetchJSON("/api/login");
  },
  async logInUser(user) {
    return await postJSON("/api/login/", user);
  },
  async createUser(user) {
    return await postJSON("/api/login/new", user);
  },
  async addOrder(order) {
    return await postJSON("/api/dish/orders", order);
  },
  async listOrders() {
    return await fetchJSON("/api/dish/orders");
  },
});
