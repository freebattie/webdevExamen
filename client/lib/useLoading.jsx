import { useEffect, useState } from "react";

export function useLoader(loadFn) {
  const [data, setData] = useState();
  const [loading, setLoading] = useState(true);
  const [error, setError] = useState();
  const reload = async () => {
    setError(undefined);
    setLoading(true);
    try {
      setData(await loadFn());
    } catch (e) {
      setError(e);
    } finally {
      setLoading(false);
    }
  };
  useEffect(() => {
    reload().catch((e) => setError(e));
  }, []);

  return { data, loading, error, reload };
}
