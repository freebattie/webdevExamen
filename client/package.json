{
  "name": "client",
  "version": "1.0.0",
  "description": "",
  "type": "module",
  "scripts": {
    "test": "jest --coverage",
    "format": "prettier --write *.{jsx,js,tsx,ts,html}",
    "dev": "parcel index.html",
    "build": "npm install --include=dev && npm run build:parcel",
    "build:parcel": "parcel build index.html"
  },

  "keywords": [],
  "author": "",
  "license": "ISC",
  "devDependencies": {
    "@babel/preset-env": "^7.19.1",
    "@babel/preset-react": "7.18.6",
    "@types/react": "^18.0.21",
    "@types/react-dom": "^18.0.6",
    "babel-jest": "^29.0.3",
    "events": "^3.3.0",
    "jest": "^29.0.3",
    "jest-environment-jsdom": "^29.0.3",
    "parcel": "^2.7.0",
    "process": "^0.11.10",
    "rimraf": "^3.0.2"
  },
  "dependencies": {
    "@parcel/transformer-react-refresh-wrap": "^2.7.0",
    "@types/jest": "^29.0.3",
    "enzyme": "^3.11.0",
    "history": "^5.3.0",
    "react": "^18.2.0",
    "react-dom": "^18.2.0",
    "react-router-dom": "^6.4.0"
  },
  "jest": {
    "testEnvironment": "jsdom",
    "collectCoverageFrom": [
      "**/*.jsx",
      "**/*.js",
      "!dist/**",
      "!coverage/**",
      "!app.jsx",
      "!index.jsx",
      "!lib/**"
    ]
  },
  "babel": {
    "presets": [
      "@babel/preset-env",
      [
        "@babel/preset-react",
        {
          "runtime": "automatic"
        }
      ]
    ]
  }
}
