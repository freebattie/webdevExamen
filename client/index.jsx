import { createRoot } from "react-dom/client";

import { App } from "./app";

// get the root element in index.html
const container = document.getElementById("app");
//creat a react root from that id
const root = createRoot(container);

root.render(<App />);
