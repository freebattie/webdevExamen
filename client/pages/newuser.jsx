import { useContext, useState } from "react";
import { Appcontext } from "../lib/appcontext.jsx";
import { useNavigate } from "react-router-dom";

export function NewUser({ setError }) {
  const [username, setUsername] = useState("");
  const [password, setPassword] = useState("");

  const [name, setName] = useState("");
  const { createUser } = useContext(Appcontext);

  const navigate = useNavigate();

  async function handleSubmit(e) {
    e.preventDefault();
    try {
      await createUser({ name, username, password });
      navigate("/");
    } catch (e) {
      setError(e);
      navigate("/errorUser");
    }
  }

  return (
    <form onSubmit={handleSubmit}>
      <div>
        <div>
          <label>
            <strong>name: </strong>
            <input
              id={name}
              type={"text"}
              value={name}
              onChange={(e) => setName(e.target.value)}
            />
          </label>
        </div>
        <div>
          <label>
            <strong>username: </strong>
            <input
              type={"text"}
              value={username}
              onChange={(e) => setUsername(e.target.value)}
            />
          </label>
        </div>
        <div>
          <label>
            <strong>password: </strong>
            <input
              type={"password"}
              value={password}
              onChange={(e) => setPassword(e.target.value)}
            />
          </label>
        </div>
      </div>
      <button className={"button"}>Save</button>
    </form>
  );
}
