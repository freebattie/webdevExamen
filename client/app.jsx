import { BrowserRouter, Routes, Route } from "react-router-dom";
import { FrontPage } from "./pages/frontpage.jsx";
import { EditUsers } from "./pages/edituser.jsx";
import { ErrorUser } from "./pages/erroruser.jsx";
import { Login } from "./pages/login.jsx";
import { NewUser } from "./pages/newuser.jsx";

import { useEffect, useState } from "react";
import { MyOrders } from "./pages/myorders.jsx";
import { Error } from "./pages/error.jsx";
import { EditMenu } from "./pages/editmenu.jsx";

import { MenuItemAdd } from "./pages/menuitemadd";
import { ChatApplication } from "./pages/chatApp.jsx";
import { Chat } from "./pages/chat";

function AddMenu({ setError }) {
  return (
    <div>
      <MenuItemAdd setError={setError} />
    </div>
  );
}

export function App() {
  const [error, setError] = useState();
  return (
    <BrowserRouter>
      <Routes>
        <Route path={"/"} element={<FrontPage setError={setError} />} />
        <Route path={"/login"} element={<Login setError={setError} />} />
        <Route path={"/register"} element={<NewUser setError={setError} />} />
        <Route path={"/edit"} element={<EditUsers />} />
        <Route path={"/error"} element={<Error error={error} />} />
        <Route path={"/errorUser"} element={<ErrorUser />} />
        <Route path={"/Myorders"} element={<MyOrders />} />
        <Route path={"/editmenu"} element={<EditMenu setError={setError} />} />
        <Route path={"/addmenu"} element={<AddMenu setError={setError} />} />
        <Route path={"/chat"} element={<Chat />} />
      </Routes>
    </BrowserRouter>
  );
}
