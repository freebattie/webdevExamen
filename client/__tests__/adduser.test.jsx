import * as React from "react";

import { createRoot } from "react-dom/client";

import { NewUser } from "../pages/newuser";

import { act, Simulate } from "react-dom/test-utils";
import { MemoryRouter } from "react-router-dom";
import { Appcontext } from "../lib/appcontext";
global.IS_REACT_ACT_ENVIRONMENT = true;
describe("add user", () => {
  it("shows form", async () => {
    const element = document.createElement("div");
    const root = createRoot(element);
    await act(() => {
      root.render(
        <MemoryRouter>
          <NewUser />
        </MemoryRouter>
      );
    });
    expect(element.innerHTML).toMatchSnapshot();

    const inputLabels = Array.from(
      element.querySelectorAll("form div div label strong")
    ).map((label) => label.innerHTML);
    expect(inputLabels).toEqual(["name: ", "username: ", "password: "]);
  });
  it("submits form", async () => {
    const element = document.createElement("div");
    const root = createRoot(element);
    const createUser = jest.fn();
    const setError = jest.fn();
    await act(async () =>
      root.render(
        <MemoryRouter>
          <Appcontext.Provider value={createUser}>
            <NewUser setError={setError} />
          </Appcontext.Provider>
        </MemoryRouter>
      )
    );

    await act(async () => {
      await Simulate.change(
        await element.querySelector("form div:nth-child(1) input"),
        {
          target: { value: "Petter petter" },
        }
      );
      await Simulate.change(
        await element.querySelector("form div:nth-child(2) input"),
        {
          target: { value: "test" },
        }
      );
      await Simulate.change(
        await element.querySelector("form div:nth-child(3) input"),
        {
          target: { value: "1234" },
        }
      );
    });
    await act(async () => {
      await Simulate.submit(await element.querySelector("form"));
    });
  });
});
