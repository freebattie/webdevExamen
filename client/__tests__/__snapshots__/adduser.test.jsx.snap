// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`add user shows form 1`] = `"<div><form><div><div><label><strong>name: </strong><input class="input" id="" type="text" value=""></label></div><div><label><strong>username: </strong><input class="input" type="text" value=""></label></div><div><label><strong>password: </strong><input class="input" type="password" value=""></label></div></div><button class="button">Save</button></form><button class="button">back</button></div>"`;
